middlewares/date.js

```diff
@@ -1,12 +1,14 @@
-var config = require('../lib/config');
+const config = require('../lib/config');
 
 module.exports = function(req, res, next) {
     res.locals.req_date = new Date();
     res.locals.true_req_date = res.locals.req_date;
 
-    // allow date override in dev mode
-    if (config.authType == 'none' && req.cookies.pl_requested_date) {
-        res.locals.req_date = new Date(Date.parse(req.cookies.pl_requested_date));
+    // We allow unit tests to override the req_date. Unit tests may also override the user
+    // (middlewares/authn.js) and the req_mode (middlewares/authzCourseOrInstance.js).
+    if (config.authType == 'none' && req.cookies.pl_test_date) {
+        res.locals.req_date = new Date(Date.parse(req.cookies.pl_test_date));
```

Comment:
The unit tests for the "active" access restriction rely on being able to simulate different times by changing the `pl_requested_date` cookie. However, it looks like this PR has removed that cookie for unit tests, and instead we have to use the `pl_test_date` cookie. I think the fix would be to go to `tests/testActiveAccessRestriction.js` and change all the lines that look like

`headers.cookie = 'pl_requested_date=_________';`

to 

`headers.cookie = 'pl_test_date=_________';`

Reply:
Thank you so much!!

It turns out that `pl_reqested_date` still exists, but it now requires authorization and can't be used by student users, which is why it fails for the "active" assessment tests (which use a student user). As you suggested, changing to `pl_test_date` was the correct fix for this.